### PR TITLE
No such file or directory with  --use_external_data_form and int8

### PR DESCRIPTION
**Pull request by rdzotz**
Noticed and issue when using convert_to_onnx.py with GPT2LMHeadModel model.
 
When using both --use_external_data_form and --precision int8 with GPT2LMHeadModel, the following error occurs
```
  File "/Users/____/miniconda3/envs/onnx/lib/python3.7/site-packages/onnx/external_data_helper.py", line 161, in save_external_data
    open(external_data_file_path, 'ab').close()
FileNotFoundError: [Errno 2] No such file or directory: '/Users/____/PycharmProjects/JDbuilder/inference/cache_models/gpt2_774M_past_int8/gpt2_774M_past_int8.onnx.data'
```

**Description**: Describe your changes.
Added the following to ensure the directory exists before running `convert_model_to_external_data` and then `onnx.save_model(quantized_onnx_model, quantized_model_path)`
`Path(os.path.split(quantized_model_path)[0]).mkdir(parents=True, exist_ok=True)`

**Motivation and Context**
Avoids the `FileNotFoundError: [Errno 2] No such file or directory:` error with the arguments --use_external_data_form and --precision int8

### CODE REVIEW

File: onnxruntime/python/tools/transformers/quantize_helper.py [hunk 1]
```diff
@@ -68,6 +68,7 @@ def quantize_onnx_model(onnx_model_path, quantized_model_path, use_external_data
 
         if use_external_data_format:
             from pathlib import Path
+            Path(os.path.split(quantized_model_path)[0]).mkdir(parents=True, exist_ok=True)
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/quantize_helper.py:71]: How about
```
Path(quantized_model_path).parent.mkdir(parents=True, exist_ok=True)
```

Comment (rdzotz) [reply onnxruntime/python/tools/transformers/quantize_helper.py:71]: That was my first attempt but I got an error parent has no mkdir error during debugging. Must have been a red herring as this works during a full run of convert_to_onnx.py

**Review by tianleiwu — APPROVED** [review]

**Comment by tianleiwu** [comment]
@rdzotz, could you integrate latest changes from master to your branch. There is a fix for CI Pipeline: ed1883a. It is needed in your branch otherwise CI Pipeline cannot pass.

**Comment by rdzotz** [comment]
@tianleiwu, this fork should now be even with the master.

**Comment by tianleiwu** [comment]
@rdzotz, I noticed that [the branch](https://github.com/rdzotz/onnxruntime/tree/patch-1) is still "8 commits behind microsoft:master". Could you integrate your fork master to that branch, then git push?

**Comment by rdzotz** [comment]
> @rdzotz, could you integrate latest changes from master to your branch? There is a fix for CI Pipeline: [ed1883a](https://github.com/microsoft/onnxruntime/commit/ed1883a97c1d73810a447b09d211bcb2e34822de). It is needed in your branch otherwise CI Pipeline cannot pass.

That branch patch:1 should now have those recent changes to the onnxruntime master

**Pull request merged**